test/translator/inquiry_test.cc

```diff
@@ -235,11 +235,54 @@ TEST(TranslateUnitSerialNumberVpd, both) {
             scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected);
   ASSERT_EQ(result.peripheral_device_type,
             scsi_defs::PeripheralDeviceType::kDirectAccessBlock);
-  ASSERT_EQ(result.page_code, 0x80);
+  ASSERT_EQ(result.page_code, scsi_defs::PageCode::kUnitSerialNumber);
   ASSERT_EQ(result.page_length, 40);
 
   // TODO: will test when response buffer is implemented
   // char formatted_hex_string[41] = "1234_5678_9abc_defa_1234_5678_9abc_defa.";
   // ASSERT_STREQ((char* )result.product_serial_number, formatted_hex_string);
 }
+
+TEST(TranslateDeviceIdentificationVpd, ReturnsCorrectFields) {
+  scsi_defs::DeviceIdentificationVpd result =
+      translator::BuildDeviceIdentificationVpd();
+    
+  ASSERT_EQ(result.peripheral_qualifier,
+      scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected);
+    
+  ASSERT_EQ(result.peripheral_device_type,
+      scsi_defs::PeripheralDeviceType::kDirectAccessBlock);
+
+  ASSERT_EQ(result.page_code, scsi_defs::PageCode::kDeviceIdentification);
+}
+
+TEST(TranslateIdentificationDescriptorNGUID, ReturnsCorrectFields) {
+  nvme_defs::IdentifyNamespace identify_namespace_data;
+  identify_namespace_data.eui64 = 0x0;
+  identify_namespace_data.nguid[0] = 0x123456789abcdefa;
+  identify_namespace_data.nguid[1] = 0x123456789abcdefa;
+  scsi_defs::IdentificationDescriptor result = 
+      translator::BuildIdentificationDescriptor(identify_namespace_data);
+  ASSERT_EQ(result.protocol_identifier, 0x0);
+  ASSERT_EQ(result.code_set, 0x1);
+  ASSERT_EQ(result.piv, 0);
+  ASSERT_EQ(result.association, 0b00);
+  ASSERT_EQ(result.identifier_type, 0x2);
+  ASSERT_EQ(result.identifier_length, 0x10);
+}
+
+TEST(TranslateIdentificationDescriptorEUI64, ReturnsCorrectFields) {
+  nvme_defs::IdentifyNamespace identify_namespace_data;
+  identify_namespace_data.eui64 = 0x123456789abcdefa;
```

Comment:
are we actually checking this value in the result?

Reply:
not the exact value but I'm checking whether `eui64` and `nguid` is zero or not